### PR TITLE
secret puller can now use a vault token or a pem

### PR DESCRIPTION
**Pull request by irwaters**
cert/key for auth.  this will allow us to migrate to useing tokens
everywhere as they are easier to invalidate

also added a new pem file for testing that doesn't expire for 10 years...

### CODE REVIEW

File: kubernets/vault-auth-secret.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@ metadata:
...
...VnVNUXd3Q2dZRFZRUUxEQU5sYm1jeEVqQVFCZ05WQkFNTQpDV3h2WTJGc2FHOXpkREVpTUNBR0NTcUdTSWIzRFFFSkFSWVRhWGRoZEdWeWMwQjZaVzVrWlhOckxtTnZiVENDCkFTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTllMdjN5RWpaSzMxOVhWODJKMFlITksKazM0WWhBNEpCV2xyRWp0dUNoVkJmOGtzQ2hUaDZ2Q2VPV1R2SVhhOU5peHVKYjVpOHovMnBlbzJzR3pGYityVwovNTQ4L1V5SFloL3RRQXZTbVRiYkxrN2NEL001cVJpcDNjdWdFOGo0OGZHNzZMVm9tTCttOGNud1RsbTZVei9kCitOUmJFOTBsM294bm54RVdpTFJuM0duRmdxakxBNVp1UzBoY0hmZlppZVZtaTlib3EyZWIvL3hxYVhacWRlM3gKaHhQQ21YbkJUVVhnTzQ1MUVXWnhMK0x1UWFIVkxrcmJaWk9WOVRDdHZuREFKaWwwaWZLa1Z0a2ZDckNxUGRiYQphRS96bktpK2RpaitRN2YwL2l2eEM3L21ZeW90V2lPMHI3UW9YcmVsdWYyNys0MUxLWjBnSWhkOWhKOVRrSmtDCkF3RUFBVEFOQmdrcWhraUc5dzBCQVFzRkFBT0NBUUVBeUJzKzZUa1NUUFhaSmM3cE9GNjdNakdBN25hdkpJRW4KTXA4MnRPejdZZE5TVXFmRm1qaGxPWFVuc1BFckJHZU9DZDVoMEpCdU5YR2J6S0hXN0hqSTFMRWxIb254dktmUQpjYmZZTzJackxLUTgvQ3RwTEx2M0JhK1B2eENBQS9sNTlHcTJRWG5lTUlqR0ZndlJLV1RvUUIydmV4TjZ6RFRUCkVMK0VQY3ZLVFVmMnBNTlJoVXhLa0JMWFlVRllacmVXSVNTUFEzY2JVT0ZYOXYwSnl3NUJvcFBjSk0yWTRKZG4KRjBoWnQvdGwrZW5sdk9oT1JmT21Hb3Y4aFJ4S3lIdkFKRHU0VnJTSzlVZDNuUy9GcmNDa092c245dTNUdElWNQo2MmlnTDVIaFRKVWk3UE1TV0l2ZFdQWlRtQjVrc21NaHZGZW1sL1NiY3NteklkWGc2OGh4Y1E9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
```
Comment (grosser) [inline kubernets/vault-auth-secret.yml]: sanity check: this is ok to be public ?

File: kubernets/vault-auth-token.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: vaultauth
+type: Opaque
+data:
+  authsecret: NTE5YzNmYWUtYTZlMy1mMDhmLTViZDAtOTI3ZGJjMmY4ZDBh
```
Comment (grosser) [inline kubernets/vault-auth-token.yml]: sanity check: this is ok to be public ?
 Add a line note
  Show notes

File: lib/secrets.rb [hunk 3]
```diff
@@ -21,17 +22,22 @@ def initialize(vault_address:, pemfile_path:, ssl_verify:, annotations:, service
...
       config.open_timeout = 3
       config.read_timeout = 2
     end
 
-    # fetch vault token by authenticating with given pem
-    response = http_post(File.join(Vault.address, CERT_AUTH_PATH), ssl_verify: ssl_verify, pem: pemfile_path)
-    Vault.token = JSON.parse(response).fetch("auth").fetch("client_token")
+    # check and see if the authfile is a pem or a token,
+    # then act accordingly
+    begin
```
Comment (grosser) [inline lib/secrets.rb]: can we check length or some other sign to know what it is ?

Comment (grosser) [inline lib/secrets.rb]: like `include?('-----BEGIN PRIVATE KEY-----')`

File: test/secrets_test.rb [hunk 4]
```diff
@@ -14,10 +14,21 @@ def process
     $stdout = old
   end
 
+  let(:token_client) do
+    SecretsClient.new(
```
Comment (grosser) [inline test/secrets_test.rb]: reuse other parameters  and merge in `authfile_path` ?

File: lib/secrets.rb (hunk 3 above)
Comment (irwaters) [inline lib/secrets.rb]: we could, but getting openssl to load on it seems like a pretty good check, no?  It'll raise the OpenSSL::X509::CertificateError if it doesn't like it.

Comment (grosser) [inline lib/secrets.rb]: yeah true no need to re-implement these checks I guess ... I'm just a little worried about broken certs being used as tokens ...

File: kubernets/vault-auth-token.yml (hunk 2 above)
Comment (irwaters) [inline kubernets/vault-auth-token.yml]: Yup, only exists on my laptop.  Really just left as an example of what the object needs to look like.  I'll add some info to the readme to that effect.

**Comment by irwaters** [comment]
@grosser when you have a sec.

**Comment by grosser** [comment]
👍

**Pull request merged**